vignettes/sorting_pruning.Rmd

```diff
@@ -143,41 +192,125 @@ Within each ethnicity separately, sort the strata by number of females in arm C
+  ) %>%
+  summarize_row_groups() %>%
+  split_rows_by("STRATA1") %>%
+  summarize_row_groups() %>%
+  analyze("AGE", afun = more_analysis_fnc)
+
+tbl <- build_table(raw_lyt, DM) %>%
+  prune_table() %>%
+  print()
+```
+
+What should we do now if we want to sort each median and mean in each of the strata variables? We need to write a custom score function as the ready-made ones at the moment work only with content nodes (`content_table()` access function for `cont_n_allcols()` and `cont_n_onecol()`, of which we will talk in a moment). But before that we need to think what are we ordering, i.e. we need to specify the right path. We suggest to look at the structure first with `table_structure()` or `row_paths_summary()`.
 
 ```{r}
-silly_afun <- function(x) {
-    in_rows(a = rcell(2),
-            b = rcell(3),
-            c = rcell(1))
+table_structure(tbl) # Direct inspection into the tree-like structure of rtables
+```
+
+We see that to order all of the `AGE` nodes we need to get there with something like this: `RACE, ASIAN, STRATA1, A, AGE` and no more as the next level is what we need to sort. But we see now that this path would sort only the first group. We need wildcards: `RACE, *, STRATA1, *, AGE`.
+
+Now, we have found a way to select relevant paths that we want to sort. We want to construct a scoring function that works on median and mean and sort them. To do so, we may want to enter our scoring function with `browser()` to see what is fed to it and try to retrieve the single value that is to be return to do the sorting. We allow the user to experiment with this, while here we show a possible solution that considers summing all the column values that are retrieved with `row_values(tt)` from the subtable that is fed to the function itself. Note that any score function should be defined as having a subtable `tt` as unique input parameter and a single numeric value as output.
+
+```{r}
+scorefun <- function(tt) {
+  # Here we could use browser()
+  sum(unlist(row_values(tt)))
 }
+sort_at_path(tbl, c("RACE", "*", "STRATA1", "*", "AGE"), scorefun)
+```
 
+To help the user visualize what is happening in the score function we show here an example of its exploration from the debugging:
+
+```
+> sort_at_path(tbl, c("RACE", "*", "STRATA1", "*", "AGE"), scorefun)
+Called from: scorefun(x)
+Browse[1]> tt ### THIS IS THE LEAF LEVEL -> DataRow ###
+[DataRow indent_mod 0]: median   30.0   33.0   36.0
+Browse[1]> row_values(tt) ### Extraction of values -> It will be a named list! ###
+$`A: Drug X`
+[1] 30
+
+$`B: Placebo`
+[1] 33
+
+$`C: Combination`
+[1] 36
+
+Browse[1]> sum(unlist(row_values(tt))) ### Final value we want to give back to sort_at_path ###
+[1] 99
+```
 
+We can see how powerful and pragmatic it might be to change the sorting principles from within the custom scoring function. We show this by selecting a specific column to sort. Looking at the pre-defined function `cont_n_onecol()` gives us an insight on how to proceed.
```

Comment:
```suggestion
We can see how powerful and pragmatic it might be to change the sorting principles from within the custom scoring function. We show this by selecting a specific column to sort. Looking at the pre-defined function `cont_n_onecol()` gives us an insight into how to proceed.
```